dali/kernels/signal/dct/dct_cpu_test.cc

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/kernels/signal/dct/dct_cpu.h"
+#include <gtest/gtest.h>
+#include <cmath>
+#include <complex>
+#include <tuple>
+#include <vector>
+#include "dali/kernels/scratch.h"
+#include "dali/kernels/common/utils.h"
+#include "dali/test/tensor_test_utils.h"
+#include "dali/test/test_tensors.h"
+
+namespace dali {
+namespace kernels {
+namespace signal {
+namespace dct {
+namespace test {
+
+template <typename T>
+void ReferenceDctTypeI(span<T> out, span<const T> in, bool normalize) {
+  int64_t in_length = in.size();
```

Comment:
Shouldn't we put similar assert as in the implementation?

Reply:
will do